### PR TITLE
mariadb 11.3.2, mariadb@11.2 11.2.3 (new formula)

### PR DESCRIPTION
**Pull request by makigumo**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----
This PR does the following:
* update mariadb latest stable version 11.3.2 (https://mariadb.com/kb/en/mariadb-11-3-2-release-notes/)
* add mariadb@11.2 11.2.3 (new formula)
* remove mariadb@11.2 alias
* add mariadb@11.3 alias

### CODE REVIEW

File: Formula/m/mariadb@11.2.rb [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+        release["release_number"]
+      end
+    end
+  end
+
+  keg_only :versioned_formula
+
+  # See: https://mariadb.com/kb/en/changes-improvements-in-mariadb-11-2/
+  # End-of-life on 2024-11-21: https://mariadb.org/about/#maintenance-policy
+  disable! date: "2024-11-21", because: :unsupported
```
Comment (chenrui333) [inline Formula/m/mariadb@11.2.rb:24]: it should be deprecate first and then disable?

File: Formula/m/mariadb@11.2.rb [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+  uses_from_macos "zlib"
+
+  on_linux do
+    depends_on "linux-pam"
+    depends_on "readline" # uses libedit on macOS
+  end
+
+  fails_with gcc: "5"
+
+  # upstream patch ref, https://github.com/MariaDB/server/pull/3064
```
Comment (chenrui333) [inline Formula/m/mariadb@11.2.rb:48]: - https://github.com/MariaDB/server/pull/3064

**Review by chenrui333 — COMMENTED** [review]
LGTM other the deprecation/disable notice for mariadb@11.2

File: Formula/m/mariadb@11.2.rb (hunk 1 above)
Comment (chenrui333) [reply Formula/m/mariadb@11.2.rb:24]: I see, sounds good to me then. 👍

**Review by carlocab — APPROVED** [review]

**Pull request merged**